core/src/main/java/org/pentaho/platform/dataaccess/datasource/wizard/csv/CsvUtils.java

```diff
@@ -462,21 +454,42 @@ public ModelInfo getModelInfo( String project, String filename ) throws FileNotF
     return (ModelInfo) xstream.fromXML( fis );
   }
 
-  private DataType convertDataType( int type ) {
+  protected int convertPrecision(  ValueMetaInterface meta  ) {
+    return meta.getPrecision() > 0 ? meta.getPrecision() : 0;
+  }
+
+  protected DataType convertDataType(  ValueMetaInterface meta  ) {
+    return convertDataType( meta.getType() );
+  }
+
+  protected DataType convertDataType( int type ) {
     switch ( type ) {
-      case 1:
-      case 5:
-      case 6:
+      case ValueMetaInterface.TYPE_NUMBER:
```

Comment:
Main reason for PR, to use the static constant variables from the ValueMetaInterface class instead of the integer values. So when the originating class ValueMetaInterface changes, this code will be automatically synchronized.